Backend/test.py

```diff
@@ -18,10 +58,56 @@
 DisplayPort over USB-C - USB-C docking station links all devices to the laptop using a single USB-C cable. Extend the traditional USB capabilities with native multi-display video, audio, data, and power delivery (on select Dell laptops) to charge the laptop with a single USB-C cable. The reversible connector is convenient to use USB-C, with no wrong orientation.
 USB 3.0 - Universal docking stations work with DisplayLink technology that enables docking features over USB. The universal docking solution enables multiple monitors, audio, Ethernet, and other USB devices to be connected to laptops through USB. Works best with USB 3.0 ports on the laptop.
 WiGig (wireless) - WiGig or wireless docking stations work with a laptop that is configured with a WiGig adapter. Only select Dell laptops support WiGig technology."""
+    # construct post request 
+    post_url = f"{endpoint}/language/authoring/analyze-text/projects/{project_name}/:import?api-version={api_version}"
+    
+    # GET request to get the status of your importing your project
+    get_url = f"{endpoint}/language/authoring/analyze-text/projects/{project_name}/import/jobs/{job_id}?api-version={api_version}"
+    
+    # Request body 
+    requests_body = "/Users/zhangc/Documents/CS Year 2/Semester 2/CSU22013 Software Engineering Project I /SwEng-Group-14-SemanticProductLink/Training_Data/weijian.json"
+   
+    # Make POST request to Azure model endpoint 
+    
+    
+    # Possible error scenarios for this request:
+    # The selected resource doesn't have proper permissions for the storage account.
+    # The storageInputContainerName specified doesn't exist.
+    # Invalid language code is used, or if the language code type isn't string.
+    # multilingual value is a string and not a boolean.
+    if request.method == "POST": 
+        response = requests.post(url=post_url, headers=headers, json=requests_body)
+        if response.status_code == 202:
+            return f"<p>Success</p>"
+        else:
+            print(response.status_code)
+            return f"<p>Fail</p>"
+    
+    else: 
+        response = requests.get(url=get_url, headers=headers)
+        return(response.get_json()) 
 
-@app.route("/")
-def hello_world():
-    return f"<p>{test_data}</p>"
+
+@app.route("/train", methods=["POST", "GET"])
```

Comment:
I see what you meant by your question in the call earlier now, adding the POST method here should work perfect👍

Reply:
Also it's the @app.route("predict") where I call the API, sorry the namings are a bit confusing

Reply:
Are you calling the API via your browser? That would be a GET request, which then explains why you receive "method not allowed" when you remove the GET as a method on your endpoint. 

Few ways you can make a POST request:
- using **curl** via command line: think this is installed by default on windows (type **curl** into a command window and it'll tell you). This page describes basics of curl: https://www.educative.io/answers/how-to-perform-a-post-request-using-curl
- via Postman: great tool which is free, it has a nice UI which is easier to work with, also lets you save requests which is handy if you're regularly testing different endpoints. Easiest to use in this case cuz you'd be attaching a body with lots of text to the request. Would look something like this 
![image](https://github.com/Ammar-tcd/SwEng-Group-14-SemanticProductLink/assets/126583060/7bea49e1-1315-487e-a85f-ff2d48d75a47)

Reply:
I'm calling the API on a function in the backend application (python file). I'm not sure if that counts as calling the API via browser? But it probably is because that explains the error. 

I think Dan got it working on Postman, so I'll ask him for that and try to transfer it to Flask! Otherwise I'll check out the package you sent in Dan's pull request

Thank you so much Sarah! 

Cora

Reply:
Ah are you referring to the Azure API? Yeah so as long as you specify POST as the method of the request in that function, that should mean it gets sent as a POST. 401 means Unauthorized so best guess is that you're missing the API key in the Headers, which you also need to specify in your code when building the request, believe it was "Ocp-Acim-Subscription-Key" as the header name, and then the header value would be whatever the key is.

Reply:
yeah the subscription key was an issue cuz someone changed it without me knowing. we got it fixed tho :))